service/history/workflow/context.go

```diff
@@ -612,15 +649,45 @@ func (c *ContextImpl) UpdateWorkflowExecutionWithNew(
 	); err != nil {
 		return err
 	}
-
-	if err := c.updateWorkflowExecutionEventReapply(
-		ctx,
-		shardContext,
-		updateMode,
-		updateWorkflowEventsSeq,
-		newWorkflowEventsSeq,
-	); err != nil {
-		return err
+	if len(updateWorkflowEventsSeq) != 0 || len(newWorkflowEventsSeq) != 0 {
```

Comment:
As commented above, I think newWorkflowEventsSeq is non-empty here. 
We need to check both xxxWorkflowEventsSeq and xxxEventReapplyCandidate and pick the non-empty one if existing for the reapply.

Reply:
updated.